src/Builder/FrontendDialectTransformer.cpp

```diff
@@ -1445,33 +1445,40 @@ void ImportFrontendModelInternal(onnx::ModelProto &model, MLIRContext &context,
   }
 }
 
-void ImportFrontendModelArray(const void *onnxBuffer, int size,
+// Return 0 on success, error otherwise.
+int ImportFrontendModelArray(const void *onnxBuffer, int size,
     MLIRContext &context, OwningOpRef<ModuleOp> &module,
-    ImportOptions options) {
+    std::string *errorMessage, ImportOptions options) {
   onnx::ModelProto model;
 
   auto parse_success = model.ParseFromArray(onnxBuffer, size);
-  assert(parse_success && "Onnx Model Parsing Failed.");
+  if (!parse_success) {
+    *errorMessage = "Unable to parse onnxBuffer";
+    return 10;
+  }
   ImportFrontendModelInternal(model, context, module, options);
+  return 0;
 }
 
-void ImportFrontendModelFile(std::string model_fname, MLIRContext &context,
+// Return 0 on success, error otherwise.
+int ImportFrontendModelFile(std::string model_fname, MLIRContext &context,
     OwningOpRef<ModuleOp> &module, std::string *errorMessage,
     ImportOptions options) {
   onnx::ModelProto model;
   std::fstream input(model_fname, std::ios::in | std::ios::binary);
   // check if the input file is opened
   if (!input.is_open()) {
     *errorMessage = "Unable to open or access " + model_fname;
-    return;
+    return 11;
   }
 
   auto parse_success = model.ParseFromIstream(&input);
   if (!parse_success) {
     *errorMessage = "Onnx Model Parsing Failed on " + model_fname;
-    return;
+    return 12;
```

Comment:
Where do the numbers 10, 11, 12 come from? Are we defining our own return codes? If that's the case, they should be defined with symbolic names.

Reply:
Was just creating random numbers... I see your point :-)